src/jira_client.ts

```diff
@@ -0,0 +1,381 @@
+import { toError } from "fp-ts/lib/Either";
+import {
+  fromEither,
+  fromLeft,
+  taskEither,
+  TaskEither,
+  tryCatch
+} from "fp-ts/lib/TaskEither";
+import * as t from "io-ts";
+import { readableReport } from "italia-ts-commons/lib/reporters";
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+import nodeFetch from "node-fetch";
+import { ServiceId } from "../generated/api/ServiceId";
+import * as config from "./config";
+
+export const JIRA_SERVICE_TAG_PREFIX = "devportal-service-";
+export const JIRA_DISABLE_LABEL = "DISATTIVAZIONE";
+
+const JIRA_CONFIG = config.getJiraConfigOrThrow();
+
+export const SearchJiraIssueResponse = t.interface({
+  startAt: t.number,
+  total: t.number,
+
+  issues: t.readonlyArray(
+    t.interface({
+      id: NonEmptyString,
+      key: NonEmptyString,
+      self: NonEmptyString,
+
+      fields: t.interface({
+        assignee: t.union([t.null, t.any]),
+        comment: t.interface({
+          comments: t.any,
+          maxResults: t.number,
+          self: t.string,
+          startAt: t.number,
+          total: t.number
+        }),
+        labels: t.union([t.null, t.any]),
+        status: t.interface({
+          name: t.string
+        }),
+        summary: t.string
+      })
+    })
+  )
+});
+export type SearchJiraIssueResponse = t.TypeOf<typeof SearchJiraIssueResponse>;
+
+export const CreateJiraIssueResponse = t.interface({
+  id: NonEmptyString,
+
+  key: NonEmptyString
+});
+export type CreateJiraIssueResponse = t.TypeOf<typeof CreateJiraIssueResponse>;
+
+export const CreateJiraCommentIssueResponse = t.interface({
+  id: NonEmptyString,
+
+  body: NonEmptyString
+  // Other properties are not relevant.
+});
+export type CreateJiraCommentIssueResponse = t.TypeOf<
+  typeof CreateJiraCommentIssueResponse
+>;
+
+const JiraIssueSearchPayload = t.interface({
+  expand: t.array(t.string),
+  fields: t.array(t.string),
+  fieldsByKeys: t.boolean,
+  jql: t.string,
+  startAt: t.number
+});
+type JiraIssueSearchPayload = t.TypeOf<typeof JiraIssueSearchPayload>;
+
+const jiraIssueSearch = (
+  baseUrl: string,
+  jiraEmail: string,
+  token: string,
+  bodyData: JiraIssueSearchPayload,
+  // tslint:disable-next-line:no-any
+  fetchApi: typeof fetch = (nodeFetch as any) as typeof fetch
+) =>
```

Comment:
If we move this method inside `JiraAPIClient` we can remove all parameters except `bodyData: JiraIssueSearchPayload`. The others parameters will be available from the parent.

Reply:
Fix with 5342287